docs/startup/preview/index.md

```diff
@@ -0,0 +1,529 @@
+
+A discarding window clears all state for the window at every emission. Each emission will only correspond to new messages that arrived since the previous emission for the window.
+
+An accumulating window retains window results from previous emissions. Each emission will contain all messages that arrived since the beginning of the window.
+
+#### Window Types:
+The Samza high-level API currently supports tumbling and session windows.
+
+**Tumbling Window**: A tumbling window defines a series of contiguous, fixed size time intervals in the stream.
+
+Examples:
+{% highlight java %}
+
+// Group the pageView stream into 3 second tumbling windows keyed by the userId
+MessageStream<PageView> pageViews = ...
+MessageStream<WindowPane<String, Collection<PageView>>> =
+                     pageViews.window(
+                         Windows.keyedTumblingWindow(pageView -> pageView.getUserId(),
+                           Duration.ofSeconds(30)))
+
+
+// Compute the maximum value over tumbling windows of 3 seconds
+MessageStream<Integer> integers = …
+Supplier<Integer> initialValue = () -> Integer.MIN_VALUE
+FoldLeftFunction<Integer, Integer> aggregateFunction = (msg, oldValue) -> Math.max(msg, oldValue)
+MessageStream<WindowPane<Void, Integer>> windowedStream =
+         integers.window(Windows.tumblingWindow(Duration.ofSeconds(30), initialValue, aggregateFunction))
+
+{% endhighlight %}
+
+
+**Session Window**: A session window groups a MessageStream into sessions. A session captures a period of activity over a MessageStream and is defined by a gap. A session is closed and results are emitted if no new messages arrive for the window for the gap duration.
+
+Examples:
+{% highlight java %}
+
+// Sessionize a stream of page views, and count the number of page-views in a session for every user
+MessageStream<PageView> pageViews = …
+Supplier<Integer> initialValue = () -> 0
+FoldLeftFunction<PageView, Integer> countAggregator = (pageView, oldCount) -> oldCount + 1;
+Duration sessionGap = Duration.ofMinutes(3);
+MessageStream<WindowPane<String, Integer> sessionCounts = pageViews.window(Windows.keyedSessionWindow(
+    pageView -> pageView.getUserId(), sessionGap, initialValue, countAggregator));
+
+// Compute the maximum value over tumbling windows of 3 seconds
+MessageStream<Integer> integers = …
+Supplier<Integer> initialValue = () -> Integer.MAX_INT
+
+FoldLeftFunction<Integer, Integer> aggregateFunction = (msg, oldValue) -> Math.max(msg, oldValue)
+MessageStream<WindowPane<Void, Integer>> windowedStream =
+     integers.window(Windows.tumblingWindow(Duration.ofSeconds(3), initialValue, aggregateFunction))
+
+{% endhighlight %}
+
+### Known Issues
+Currently, both window and join operators buffer messages in-memory. So, messages could be lost on failures and re-starts.
```

Comment:
I couldn't find one, though I'm sure it exists. @prateekm @vjagadish1989 do you happen to know? And do you think we should link to the jiras for each known issue?